pkg/controller/mutators/core/reconciler.go

```diff
@@ -141,7 +142,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	// for conflicts.
 	r.queueConflicts(diff)
 
-	ingestionStatus = ctrlmutators.MutatorStatusActive
+	// Any mutator that's in conflict with another should be in the "error" state.
```

Comment:
why do we default to error state assuming there is conflict?

Reply:
Please see this [other comment thread](https://github.com/open-policy-agent/gatekeeper/pull/1714#discussion_r761306757) where we discussed how conflicting errors should be reported.

Do you see a better way the mutators could describe the state of the system?  If so, could you outline it here?

Reply:
The thread talks about conflict vs error as separate metrics. My question above is more why do we default to error state assuming there is a conflict?

Reply:
Perhaps I'm misunderstanding your question, but I think what we're trying to convey is that a mutator is in an `active` state only if it is:

1. Successfully within the mutation system (ingested)
2. Not in conflict with another mutator

Thus, in this logic, we default to `error` and check for successful ingestion (do we get past the `reconcile...` function calls) and conflicts before setting the status to `MutatorStatusActive`.  

Is that what you were asking about?

Reply:
Wait, I think I see what you're talking about now.  

What if a mutator fails due to an error, but isn't in conflict?  The current logic would report it as a conflict.  That is unintentional.

Will change!

Reply:
@ritazh PTAL at the new logic

Reply:
good catch, that was subtle.